cli-output/src/cli_output.rs

```diff
@@ -1142,3 +1147,147 @@ impl fmt::Display for CliFees {
         Ok(())
     }
 }
+
+pub fn return_signers(
+    tx: &Transaction,
+    output_format: &OutputFormat,
+) -> Result<String, Box<dyn std::error::Error>> {
+    let verify_results = tx.verify_with_results();
+    let mut signers = Vec::new();
+    let mut absent = Vec::new();
+    let mut bad_sig = Vec::new();
+    tx.signatures
+        .iter()
+        .zip(tx.message.account_keys.iter())
+        .zip(verify_results.into_iter())
+        .for_each(|((sig, key), res)| {
+            if res {
+                signers.push(format!("{}={}", key, sig))
+            } else if *sig == Signature::default() {
+                absent.push(key.to_string());
+            } else {
+                bad_sig.push(key.to_string());
+            }
+        });
+
+    let cli_command = CliSignOnlyData {
+        blockhash: tx.message.recent_blockhash.to_string(),
+        signers,
+        absent,
+        bad_sig,
+    };
+
+    Ok(output_format.formatted_string(&cli_command))
+}
+
+#[allow(clippy::type_complexity)]
+pub fn parse_sign_only_reply_string(
+    reply: &str,
+) -> (Hash, Vec<(Pubkey, Signature)>, Vec<Pubkey>, Vec<Pubkey>) {
```

Comment:
I'm pretty sure you don't need this workaround. This can just return a `solana_clap_utils::keypair::SignOnly`.

`clap-utils` is at the "bottom"
`client` depends on `clap-utils`
`cli-output` depends on `client` and `clap-utils`
`cli` depends on `cli-output`, `client`, and `clap-utils`

Reply:
Oh nice!  I'll give it a go.  The clippy-allow made me sad 🙁

Reply:
Yay!  Much better https://github.com/solana-labs/solana/pull/12382/commits/acb933655360f753b5a0645215ef167b9eac36aa

Good eye!